airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_sources.py

```diff
@@ -64,10 +64,15 @@ def get_dag_source(
             status.HTTP_404_NOT_FOUND,
             f"The source code of the DAG {dag_id}, version_number {version_number} was not found",
         )
-
-    dag_source = dag_version.dag_code.source_code
+    if dag_code := dag_version.dag_code:
+        content = dag_code.source_code
+    else:
+        content = f"# Code not found for dag '{dag_id}'"
+        if version_number:
+            content += f" and version '{version_number}'"
```

Comment:
This feels slightly wierd to not be served as a 404 error, but I get why we are doing it this way.

I still don't like it.

Reply:
If this was in the UI only route, this would be fine, but I'm really not convinced it should be a 200 in the public REST api.

Reply:
ok i have made it 404